### PR TITLE
F3 02 chrysics

### PR DESCRIPTION
**Pull request by CHRYSICS**
Work towards streaming the ethical ingredients options (in progress). Includes ideas from prototype, such as recipe profile.
Here, changed search window to make it clear that user is looking in database for "common recipes"
progress1

Here, view template is close to just letting the user view the common searched/build recipe in more detail, navigates to "update":
progress2

Here, update page is now set up with viewing of alternatives and a recipe profile rating table.  This is is still a work in progress. The hope is that now it allow the alternatives be a "select" like feature that update the page as a POST request (not there yet).  Additionally, the recipe after modifying will navigate user to save recipe in user "cookbook" database, not yet created.  If we create a user cookbook database to store user recipes, then modification will not over write common recipes found on main database.  
progress3

Note, I create cookbook.py to store the orginal update/delete routes from the recipe.py so we can adapt them for the cookbook database. Additionally, the original update view is now stored as the "edit_recipe.html" in the cookbook templates for reference/usage in cookbook templates.
progress4
progress5

Sorry about the huge dump, got kind of lost in javascript and didn't realize all I changed until afterwards.
-cheers

### CODE REVIEW

**Comment by CHRYSICS** [comment]
STILL IN PROGRESS
Made available for progress since this is larger than I was hoping

**Comment by elaguerta** [comment]
I'm in the middle of testing. It's beautiful! I found one bug that I don't think is a blocker for merging, but I'll comment here and file an issue in a minute. When swapping ingredients, the colored ratings in the table don't seem to change to reflect the newly selected ingredient. For example, I swapped to 'butter-organic', but the `r_global_warming = 2.0` still shows up under Cons: 
image

File: example/flaskr/static/js/ratingTable.js [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+                delete ingredients[item];
+
+                ingredients[name] = {};
+                ingredients[name]["recipe_id"] = recipe_id;
+                ingredients[name]["ingredient_id"] = selected.value;
+                ingredients[name]["category"] = category;
+                ingredients[name]["amount"] = amount;
+                ingredients[name]["unit"] = unit;
+                calcSubRatings(ingredients, alts);
+                calcCategoryRatings();
```
Comment (elaguerta) [inline example/flaskr/static/js/ratingTable.js:301]: Need to add statements here to also change the highlighted ratings in the 'Pros' and 'Cons' column.

**Review by elaguerta — APPROVED** [review]
This is amazing work! The javascript alone makes this portfolio ready imo. Left a comment in line about adding some logic to handle the updating of the 'Pros' and 'Cons'  columns, which we can save for another PR or flag as a bug for our homework. Great job!

File: example/flaskr/static/js/ratingTable.js (hunk 1 above)
Comment (CHRYSICS) [reply example/flaskr/static/js/ratingTable.js:301]: Thanks, that will be to next PR update! :)
Will merge for now and continue working on the pros/cons column! :+1:

**Pull request merged**